python/ray/serve/metric/sink.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict
+from collections import Counter, namedtuple
+
+import ray
+from ray.serve.metric.types import MetricType, MetricMetadata, MetricBatch
+from ray.serve.utils import _get_logger
+
+logger = _get_logger()
+
+
+def make_metric_namedtuple(metadata: MetricMetadata, record: MetricBatch):
+    fields = ["name", "type"]
+    fields += list(metadata.default_labels.keys())
+    fields += list(record.labels.keys())
+
+    merged_labels = metadata.default_labels.copy()
+    merged_labels.update(record.labels)
+
+    tuple_type = namedtuple(metadata.name, fields)
+    return tuple_type(name=metadata.name, type=metadata.type, **merged_labels)
+
+
+class BaseSink:
+    """A sink is the place to store or forward metrics to external services"""
+
+    def __init__(self):
+        self.metadata = dict()
+        self.recording = []
+        logger.debug("Sink initialized {}".format(type(self)))
+
+    def push_batch(self, metadata: Dict[str, MetricMetadata],
+                   batch: MetricBatch):
+        self.metadata.update(metadata)
+        self.recording.extend(batch)
+        self.compact()
+
+    def compact(self):
+        raise NotImplementedError(
+            "This method should be implemented by subclass.")
+
+    def get_metric(self):
```

Comment:
```suggestion
    def get_metrics(self):
```

Reply:
It seems sort of strange to me that both this and `compact` would be implemented by the same class. One is storing metrics and serving them, one is just pushing them out. Doesn't seem like the concepts should be combined in one interface.

Reply:
changed this to `inspect_metrics` which makes more sense since this method will only be called by `serve.stat` or `/-/metrics`